awx_collection/tests/integration/targets/tower_credential_input_source/tasks/main.yml

```diff
@@ -27,17 +27,17 @@
     inputs:
       username: user
     organization: Default
-  register: result
+  register: target_cred_result
 
 - assert:
     that:
-      - "result is changed"
+      - "target_cred_result is changed"
 
 - name: Add credential Input Source
   tower_credential_input_source:
     input_field_name: password
-    target_credential: "{{ target_cred_name }}"
-    source_credential: "{{ src_cred_name }}"
+    target_credential: "{{ target_cred_result.id }}"
+    source_credential: "{{ src_cred_result.id }}"
```

Comment:
right below this we seem to be getting and error:

```
  PLAY [testhost] ****************************************************************
  
  TASK [Gathering Facts] *********************************************************
  ok: [testhost]
  
  TASK [tower_credential_input_source : Generate names] **************************
  ok: [testhost]
  
  TASK [tower_credential_input_source : Add Tower credential Lookup] *************
  changed: [testhost]
  
  TASK [tower_credential_input_source : assert] **********************************
  ok: [testhost] => {
      "changed": false, 
      "msg": "All assertions passed"
  }
  
  TASK [tower_credential_input_source : Add Tower credential Target] *************
  changed: [testhost]
  
  TASK [tower_credential_input_source : assert] **********************************
  ok: [testhost] => {
      "changed": false, 
      "msg": "All assertions passed"
  }
  
  TASK [tower_credential_input_source : Add credential Input Source] *************
  changed: [testhost]
  
  TASK [tower_credential_input_source : assert] **********************************
  fatal: [testhost]: FAILED! => {"msg": "The conditional check 'result is changed' failed. The error was: The 'changed' test expects a dictionary"}
  
  PLAY RECAP *********************************************************************
  testhost                   : ok=7    changed=3    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   
```

Maybe this task is missing a `register:`?

Maybe you missed a rename `result` -> `target_cred_result`?

Reply:
It was missing a register. I also updated this test to be more friendly while I was in there.